### ISSUE TITLE
fetch from Bun server get stuck forever

### ISSUE THREAD

**Issue by hanabi1224** [oven-sh/bun#1196]
### Version

0.1.10

### Platform

Linux 5.10.102.1-microsoft-standard-WSL2 (ubuntu 20.04)

### What steps will reproduce the bug?

Run below code with `bun run app.js 10`

```javascript
const host = "localhost";

async function send(api, value) {
	let r = await fetch(api, {
		method: "POST",
		body: JSON.stringify({ value: value }),
	});
	let text = await r.text();
	console.log(`response: ${text}`);
	return parseInt(text);
}

async function sendAsync(api, value) {
	while (true) {
		try {
			let v = await send(api, value);
			return v;
		} catch (e) {
			console.log(e);
		}
	}
}

async function calculateSum(port, n) {
	const api = `http://${host}:${port}/`;
	var sum = +0;
	var tasks = [];
	for (var i = 1; i <= n; i++) {
		tasks.push(sendAsync(api, i));
	}
	for (var i = 0; i < n; i++) {
		sum += await tasks[i];
	}
	console.log(sum);
}

function runServer(port) {
	return Bun.serve({
		async fetch(req) {
			if (req.method === "POST") {
				const payload = await req.json();
				console.log(`request: ${payload.value}`);
				return new Response(payload.value.toString());
			} else {
				return new Response(`${req.method} not supported`);
			}
		},
		error(err) {
			return new Response("uh oh! :(\n" + err.toString(), { status: 500 });
		},
		development: false,
		port: port,
	});
}

async function main() {
	const args = process.argv.slice(2);
	const n = +args[0] || 10;
	const port = 20000 + Math.floor(Math.random() * 30000);
	const server = runServer(port);
	await calculateSum(port, n);
	server.stop();
}

main();
```

### How often does it reproduce? Is there a required condition?

It succeeds 1 out of maybe 10 runs, but in most cases just struck there.

### What is the expected behavior?

It should print out an integer and quit.

When I run a similar js program with `node app.js 1000`, it never stucks

```javascript
const http = require("http");

const host = "localhost";

function requestListener(req, res) {
	var body = "";
	req.on("data", function (chunk) {
		body += chunk;
	});
	req.on("end", function () {
		const payload = JSON.parse(body);
		res.writeHead(200);
		res.end(payload.value.toString());
	});
}

function send(api, value) {
	return new Promise(function (onFulfilled, onRejected) {
		const req = http.request(
			api,
			{
				method: "POST",
				headers: {
					"content-type": "application/json",
				},
			},
			(res) => {
				var content = "";
				res.on("data", (chunk) => {
					content += chunk;
				});
				res.on("end", () => {
					onFulfilled(+content);
				});
			},
		);
		req.on("error", (err) => {
			onRejected(err);
		});
		req.write(JSON.stringify({ value: value }));
		req.end();
	});
}

async function sendAsync(api, value) {
	while (true) {
		try {
			return await send(api, value);
		} catch (e) {}
	}
}

async function calculateSum(port, n) {
	const api = `http://${host}:${port}/`;
	var sum = +0;
	var tasks = [];
	for (var i = 1; i <= n; i++) {
		tasks.push(sendAsync(api, i));
	}
	for (var i = 0; i < n; i++) {
		sum += await tasks[i];
	}
	console.log(sum);
}

function main() {
	const args = process.argv.slice(2);
	const n = +args[0] || 10;
	const port = 20000 + Math.floor(Math.random() * 30000);
	const server = http.createServer(requestListener);
	server.listen(port, host, async () => {
		await calculateSum(port, n);
		server.close();
	});
}

main();
```

### What do you see instead?

The bun version gets stuck in most cases even with a very small input (<10)

**Comment by fabiancook** [comment]
I think this may be the same as https://github.com/oven-sh/bun/issues/1190 

Do you get `response.status: 0` too? 

I also found that bun isn't exiting with code 1 if there is an unhandled rejection / top level await rejection, [which would make it look like the process was just hanging](https://github.com/virtualstate/focus/runs/8111248985?check_suite_focus=true#step:10:1268) 

So I would use: 

```javascript
await main().catch(error => {
    console.error(error);
    process.exit(1);
})
```

Its interesting you also use `const port = 20000 + Math.floor(Math.random() * 30000);`, is this the same as what node/deno actually uses?

I had come across this problem too where with deno & node I would be able to use `port: 0` and get a random port, but get this with bun, which is fair:

![image](https://user-images.githubusercontent.com/4337080/188118927-67772282-2071-402d-ad08-22351edea715.png)

**Comment by Jarred-Sumner** [comment]
Should be fixed as of Bun v0.1.13 or Bun v0.2.0
